include/aspect/simulator.h

```diff
@@ -95,248 +95,12 @@ namespace aspect
   class Simulator
   {
     private:
-      /**
-       * A structure that contains enum values that identify the nonlinear
-       * solver in use.
-       */
-      struct NonlinearSolver
-      {
-        enum Kind
-        {
-          IMPES,
-          iterated_IMPES,
-          iterated_Stokes,
-          Stokes_only
-        };
-      };
 
-      struct NullspaceRemoval
-      {
-        enum Kind
-        {
-          none = 0,
-          net_translation_x = 0x1,
-          net_translation_y = 0x2,
-          net_translation_z = 0x4,
-          net_translation   = 0x1+0x2+0x4,
-          linear_momentum_x = 0x8,
-          linear_momentum_y = 0x10,
-          linear_momentum_z = 0x20,
-          linear_momentum   = 0x8+0x10+0x20,
-          net_rotation      = 0x40,
-          angular_momentum  = 0x80
-        };
-      };
+      typedef typename Parameters<dim>::NonlinearSolver NonlinearSolver;
+      typedef typename Parameters<dim>::NullspaceRemoval NullspaceRemoval;
```

Comment:
we basically had the same before:
```
 typedef typename NonlinearSolver::Kind NonlinearSolverKind;
```
but I guess we can get rid of this. Will do...